viam-cartographer/src/carto_facade/map_builder.h

```diff
@@ -28,6 +28,7 @@ const SensorId kOdometerSensorId{SensorId::SensorType::ODOMETRY, "odometry"};
 
 class MapBuilder {
    public:
+    MapBuilder() : trajectory_builder(){};
```

Comment:
[q] what does adding this here do?

Reply:
[The valgrind test was complaining](https://github.com/viamrobotics/viam-cartographer/actions/runs/7411106720/job/20166615104#step:14:3468) that `trajectory_builder` was not initialized [but used in an if-statement](https://github.com/kkufieta/viam-cartographer/blob/kk/rsdk-5334-remove-unused-code/viam-cartographer/src/carto_facade/map_builder.cc#L28) (`Conditional jump or move depends on uninitialised value(s)`), which [happens when we error out here](https://github.com/kkufieta/viam-cartographer/blob/kk/rsdk-5334-remove-unused-code/viam-cartographer/src/carto_facade/carto_facade.cc#L237).